sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -81,6 +81,8 @@ class Analyzer(
       UnresolvedHavingClauseAttributes ::
       HiveTypeCoercion.typeCoercionRules ++
       extendedResolutionRules : _*),
+    Batch("Cleanup", fixedPoint,
```

Comment:
We remove subqueries in optimizer, but `UnresolvedAlias` is another story, we have to strip(resolve) it before `CheckAnalysis`.